src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.StateManager.HostStates.cs

```diff
@@ -69,16 +70,13 @@ private int GetPriority(StateSet state)
                         return BuiltInCompilerPriority;
                     }
 
-                    switch (state.Analyzer)
+                    return state.Analyzer switch
                     {
-                        case DocumentDiagnosticAnalyzer analyzer:
-                            return Math.Max(0, analyzer.Priority);
-                        case ProjectDiagnosticAnalyzer analyzer:
-                            return Math.Max(0, analyzer.Priority);
-                        default:
-                            // regular analyzer get next priority after compiler analyzer
-                            return RegularDiagnosticAnalyzerPriority;
-                    }
+                        FileContentLoadAnalyzer _ => FileContentLoadAnalyzerPriority,
+                        DocumentDiagnosticAnalyzer analyzer => Math.Max(0, analyzer.Priority),
+                        ProjectDiagnosticAnalyzer analyzer => Math.Max(0, analyzer.Priority),
+                        _ => RegularDiagnosticAnalyzerPriority,
+                    };
```

Comment:
📝 These refactorings mixed with code changes are substantially increasing the amount of work required to review these already-large changes. It would help tremendously to save them all for another time.

Reply:
I made most of the refactorings in the separate PR. There is a balance of how much refactoring to do separately vs. with changes. Moving all refactorings to separate PR also doesn't scale.